Plugins/Public/base_plugin/CoreModule.cpp

```diff
@@ -410,163 +410,137 @@ void CoreModule::SaveState(FILE *file)
-				base->base_health -= damage_taken;
+				base->RemoveMarketGood(i.first, number_of_people);
 			}
-
-			// Repair damage if we have sufficient crew on the base.
-			base->repairing = false;
-			if (isCrewSufficient) {
-				RepairDamage(base->max_base_health);
-				if (dont_eat) {
-					// We won't save base health below, so do it here
-					float rhealth = base->base_health / base->max_base_health;
-					pub::SpaceObj::SetRelativeHealth(space_obj, rhealth);
-				}
+			else {
+				base->RemoveMarketGood(i.first, base->HasMarketItem(i.first));
+				passedCheck = false;
 			}
+		}
 
-			if (base->base_health > base->max_base_health)
-				base->base_health = base->max_base_health;
-			else if (base->base_health <= 0)
-				base->base_health = 0;
+		// Humans use food but may eat one of a number of types.
 
-			if (!dont_eat)
-			{
-				// Humans use commodity_oxygen, commodity_water. Consume these for
-				// the crew or kill 10 crew off and repeat this every 12 hours.
-				if (time % 43200 == 0)
-				{
-					for (map<uint, uint>::iterator i = set_base_crew_consumption_items.begin();
-						i != set_base_crew_consumption_items.end(); ++i)
-					{
-						// Use water and oxygen.
-						if (base->HasMarketItem(i->first) >= number_of_crew)
-						{
-							base->RemoveMarketGood(i->first, number_of_crew);
-						}
-						// Insufficient water and oxygen, kill crew.
-						else
-						{
-							base->RemoveMarketGood(set_base_crew_type, (number_of_crew >= 10) ? 10 : number_of_crew);
-						}
-					}
-
-					// Humans use food but may eat one of a number of types.
-					uint crew_to_feed = number_of_crew;
-					for (map<uint, uint>::iterator i = set_base_crew_food_items.begin();
-						i != set_base_crew_food_items.end(); ++i)
-					{
-						if (!crew_to_feed)
-							break;
-
-						uint food_available = base->HasMarketItem(i->first);
-						if (food_available)
-						{
-							uint food_to_use = (food_available >= crew_to_feed) ? crew_to_feed : food_available;
-							base->RemoveMarketGood(i->first, food_to_use);
-							crew_to_feed -= food_to_use;
-						}
-					}
-
-					// Insufficent food so kill crew.
-					if (crew_to_feed)
-					{
-						base->RemoveMarketGood(set_base_crew_type, (crew_to_feed >= 10) ? 10 : crew_to_feed);
-					}
-				}
-
-				// Save the new base health
-				float rhealth = base->base_health / base->max_base_health;
-				pub::SpaceObj::SetRelativeHealth(space_obj, rhealth);
-				if (set_plugin_debug > 1)
-					ConPrint(L"CoreModule::timer space_obj=%u health=%f\n", space_obj, base->base_health);
+		for (auto& i : set_base_crew_food_items)
+		{
+			if (!number_of_people)
+				break;
 
+			uint food_available = base->HasMarketItem(i.first);
+			if (food_available)
+			{
+				uint food_to_use = min(food_available, number_of_people);
+				base->RemoveMarketGood(i.first, food_to_use);
+				number_of_people -= food_to_use;
```

Comment:
how about we name this `unfed_people`

Reply:
peopleToFeed made more sense to me, added in next commit